docs/development/zod.md

```diff
@@ -0,0 +1,355 @@
+### Table of Contents
+
+- [Zod schema guideline](#zod-schema-guideline)
+  - [When and where to use Zod](#when-and-where-to-use-zod)
+  - [Technical guide](#technical-guide)
+    - [Use `schema.ts` files for Zod schemas](#use-schemats-files-for-zod-schemas)
+    - [Name schemas without any `Schema` suffix](#name-schemas-without-any-schema-suffix)
+    - [Inferred types](#inferred-types)
+    - [Specify only necessary fields](#specify-only-necessary-fields)
+    - [Use `Json`, `Yaml` and `Toml` for string parsing](#use-json-yaml-and-toml-for-string-parsing)
+    - [Use `.transform()` method to process parsed data](#use-transform-method-to-process-validated-data)
+      - [Rename and move fields at the top level transform](#rename-and-move-fields-at-the-top-level-transform)
+    - [Stick to permissive behavior when possible](#stick-to-permissive-behavior-when-possible)
+      - [Use `.catch()` to force default values](#use-catch-to-force-default-values)
+      - [Use `LooseArray` and `LooseRecord` to filter out incorrect values from collections](#use-loosearray-and-looserecord-to-filter-out-incorrect-values-from-collections)
+    - [Combining with `Result` class](#combining-with-result-class)
+    - [Combining with `Http` class](#combining-with-http-class)
```

Comment:
```suggestion
If you're viewing this file on GitHub, you can use a special button to see a auto-generated table of contents.
Read the [GitHub Docs, Basic writing and formatting syntax, headings](https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#headings) section to learn more about this button.
```

The GitHub Docs shows what the "table of contents button and icon" looks like. So we can drop the manual TOC and rely on GitHub's TOC.

I don't know if we need to say: "If you're using a text editor you can probably get a table of contents or overview tab." It feels too obvious to mention this.

Reply:
I'm afraid that almost nobody actually uses it. I noticed this feature right in the moment of me searching how to make Markdown TOC for GitHub.